# Proyecto de Ciencia de Datos - Universidades de Chile

`Item I:` Limpieza de datos.

`Item II:` Estadísticos descriptivos.

`Item III:` Tabla de contingencia.

`Item IV:` Uso Interpretación de gráficos.

`Item V:` Estructura del informe, ortografía, redacción y calidad gráfica.


In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Opciones para evitar notación científica en NumPy y Pandas
np.set_printoptions(suppress=True)
pd.set_option('display.float_format', lambda x: f"{x:0.5f}")

# Opciones de gráficos
plt.style.use("ggplot")
%config InlineBackend.figure_format = "retina"

In [2]:
matriculados = pd.read_csv('data/OFICIAL_WEB_PROCESO_MAT_2007_AL_2022_24_06_2022_E.csv', encoding='latin1',sep=';')

C:\Users\aigna\AppData\Local\Temp\ipykernel_24328\2471375057.py:1: DtypeWarning: Columns (27) have mixed types. Specify dtype option on import or set low_memory=False.
  matriculados = pd.read_csv('data/OFICIAL_WEB_PROCESO_MAT_2007_AL_2022_24_06_2022_E.csv', encoding='latin1',sep=';')


In [3]:
print("Base de Datos de Matriculados:")
display(matriculados.head())

Base de Datos de Matriculados:


AÑO  TOTAL MATRICULADOS  MATRICULADOS MUJERES POR PROGRAMA  \
0  MAT_2022                   2                            2.00000   
1  MAT_2022                  10                            8.00000   
2  MAT_2022                  34                                NaN   
3  MAT_2022                  10                                NaN   
4  MAT_2022                  76                            4.00000   

   MATRICULADOS HOMBRES POR PROGRAMA  TOTAL MATRICULADOS PRIMER AÑO  \
0                                NaN                            NaN   
1                            2.00000                            NaN   
2                           34.00000                            NaN   
3                           10.00000                            NaN   
4                           72.00000                            NaN   

   MATRICULADOS MUJERES PRIMER AÑO  MATRICULADOS HOMBRES PRIMER AÑO  \
0                              NaN                              NaN   
1                              NaN                              NaN   
2                              NaN                              NaN   
3                              NaN                              NaN   
4                              NaN                              NaN   

  CLASIFICACIÓN INSTITUCIÓN NIVEL 1 CLASIFICACIÓN INSTITUCIÓN NIVEL 2  \
0          Institutos Profesionales          Institutos Profesionales   
1          Institutos Profesionales          Institutos Profesionales   
2          Institutos Profesionales          Institutos Profesionales   
3          Institutos Profesionales          Institutos Profesionales   
4          Institutos Profesionales          Institutos Profesionales   

  CLASIFICACIÓN INSTITUCIÓN NIVEL 3  ...  TES PARTICULAR SUBVENCIONADO  \
0          Institutos Profesionales  ...                       2.00000   
1          Institutos Profesionales  ...                       3.00000   
2          Institutos Profesionales  ...                      14.00000   
3          Institutos Profesionales  ...                       4.00000   
4          Institutos Profesionales  ...                      29.00000   

  TES PARTICULAR PAGADO TES CORP. DE ADMINISTRACIÓN DELEGADA  \
0                   NaN                                  NaN   
1               2.00000                                  NaN   
2               3.00000                                  NaN   
3                   NaN                                  NaN   
4               6.00000                              4.00000   

  TES SERVICIO LOCAL EDUCACION TOTAL TES % COBERTURA TES  \
0                          NaN         2          100,0%   
1                          NaN         9           90,0%   
2                          NaN        26           76,5%   
3                          NaN        10          100,0%   
4                          NaN        71           93,4%   

  TIPO ESTABLECIMIENTO HC TIPO ESTABLECIMIENTO TP CLAS_EST ADULTO  \
0                 2.00000                     NaN             NaN   
1                 4.00000                 5.00000         2.00000   
2                17.00000                 9.00000         4.00000   
3                 6.00000                 4.00000         1.00000   
4                30.00000                41.00000         4.00000   

  CLAS_EST JOVEN  
0        2.00000  
1        7.00000  
2       22.00000  
3        9.00000  
4       67.00000  

[5 rows x 56 columns]

In [4]:
print("Identificación de Datos Faltantes")
display(matriculados.isna().sum()[matriculados.isna().sum()>0])

Identificación de Datos Faltantes


MATRICULADOS MUJERES POR PROGRAMA        16242
MATRICULADOS HOMBRES POR PROGRAMA        17063
TOTAL MATRICULADOS PRIMER AÑO            64261
MATRICULADOS MUJERES PRIMER AÑO          75937
MATRICULADOS HOMBRES PRIMER AÑO          76222
MODALIDAD                                42300
DURACIÓN ESTUDIO CARRERA                   106
DURACIÓN TOTAL DE CARRERA                  106
RANGO DE EDAD 15 A 19 AÑOS               99919
RANGO DE EDAD 20 A 24 AÑOS               41284
RANGO DE EDAD 25 A 29 AÑOS               21579
RANGO DE EDAD 30 A 34 AÑOS               44593
RANGO DE EDAD 35 A 39 AÑOS               82881
RANGO DE EDAD 40 Y MÁS AÑOS              93610
RANGO DE EDAD SIN INFORMACIÓN           213170
 PROMEDIO EDAD CARRERA                      39
 PROMEDIO EDAD MUJER                     16287
 PROMEDIO EDAD HOMBRE                    17113
TES MUNICIPAL                            32208
TES PARTICULAR SUBVENCIONADO             28069
TES PARTICULAR PAGADO                   103756
TES CORP. DE 

In [50]:
matriculados.isna().sum()[matriculados.isna().sum()>0].reset_index()

index       0
0      MATRICULADOS MUJERES POR PROGRAMA   16242
1      MATRICULADOS HOMBRES POR PROGRAMA   17063
2          TOTAL MATRICULADOS PRIMER AÑO   64261
3        MATRICULADOS MUJERES PRIMER AÑO   75937
4        MATRICULADOS HOMBRES PRIMER AÑO   76222
5                              MODALIDAD   42300
6               DURACIÓN ESTUDIO CARRERA     106
7              DURACIÓN TOTAL DE CARRERA     106
8             RANGO DE EDAD 15 A 19 AÑOS   99919
9             RANGO DE EDAD 20 A 24 AÑOS   41284
10            RANGO DE EDAD 25 A 29 AÑOS   21579
11            RANGO DE EDAD 30 A 34 AÑOS   44593
12            RANGO DE EDAD 35 A 39 AÑOS   82881
13           RANGO DE EDAD 40 Y MÁS AÑOS   93610
14         RANGO DE EDAD SIN INFORMACIÓN  213170
15                PROMEDIO EDAD CARRERA       39
16                  PROMEDIO EDAD MUJER    16287
17                 PROMEDIO EDAD HOMBRE    17113
18                         TES MUNICIPAL   32208
19          TES PARTICULAR SUBVENCIONADO   28069
20                 TES PARTICULAR PAGADO  103756
21  TES CORP. DE ADMINISTRACIÓN DELEGADA  113934
22          TES SERVICIO LOCAL EDUCACION  208336
23               TIPO ESTABLECIMIENTO HC   20156
24               TIPO ESTABLECIMIENTO TP   42829
25                       CLAS_EST ADULTO   76329
26                        CLAS_EST JOVEN   15555

## Análisis 1: Verificación de columnas Matriculadas Mujeres y Hombres por programa.

Se verificará si las columnas `MATRICULADOS MUJERES POR PROGRAMA` y `MATRICULADOS HOMBRES POR PROGRAMA` con la columna de `TOTAL MATRICULADOS`.



In [5]:
print("Total Matriculados: ")
display(matriculados['TOTAL MATRICULADOS'].sum().astype(int))
print("Total Matriculadas Mujeres: ")
display(matriculados['MATRICULADOS MUJERES POR PROGRAMA'].sum().astype(int))
print("Total Matriculados Hombres: ")
display(matriculados['MATRICULADOS HOMBRES POR PROGRAMA'].sum().astype(int))



Total Matriculados: 


18146411

Total Matriculadas Mujeres: 


9481945

Total Matriculados Hombres: 


8664466

In [6]:
print("La sumatoria entre el total de matriculadas mujeres y hombres es:")
print(matriculados['MATRICULADOS MUJERES POR PROGRAMA'].sum().astype(int) + matriculados['MATRICULADOS HOMBRES POR PROGRAMA'].sum().astype(int))

La sumatoria entre el total de matriculadas mujeres y hombres es:
18146411


In [7]:
print("Se verifica a través de una igualdad si la sumatoria total corresponde al total de matriculados:")
print(matriculados['MATRICULADOS MUJERES POR PROGRAMA'].sum().astype(int) + matriculados['MATRICULADOS HOMBRES POR PROGRAMA'].sum().astype(int) == matriculados['TOTAL MATRICULADOS'].sum().astype(int))

Se verifica a través de una igualdad si la sumatoria total corresponde al total de matriculados:
True


### Conclusión 1

Se verifica que la sumatoria total de Matriculados entre mujeres y hombres resulta en la igualdad del total de matriculados de la base de datos, por ende los datos de las columnas `MATRICULADOS MUJERES POR PROGRAMA` y `MATRICULADOS HOMBRES POR PROGRAMA` se pueden reemplazar por 0. para eliminar de la base valores faltantes.

## Analisis 2

Se analizan las columnas: `TOTAL MATRICULADOS PRIMER AÑO` con `MATRICULADOS MUJERES PRIMER AÑO` Y `MATRICULADOS HOMBRES PRIMER AÑO`

In [8]:
print("Total Matriculados de primer año:")
print(matriculados[~matriculados['TOTAL MATRICULADOS PRIMER AÑO'].isna()]['TOTAL MATRICULADOS PRIMER AÑO'].sum().astype(int))
print("Total Matriculadas Mujeres de primer año:")
print(matriculados[~matriculados['MATRICULADOS MUJERES PRIMER AÑO'].isna()]['MATRICULADOS MUJERES PRIMER AÑO'].sum().astype(int))
print("Total Matriculados Hombres de primer año")
print(matriculados[~matriculados['MATRICULADOS HOMBRES PRIMER AÑO'].isna()]['MATRICULADOS HOMBRES PRIMER AÑO'].sum().astype(int))


Total Matriculados de primer año:
5441704
Total Matriculadas Mujeres de primer año:
2825386
Total Matriculados Hombres de primer año
2616318


In [9]:
print("Se establecen dos condiciones para verificar la integridad de los datos:")
print("1.- Se filtra por todos los matriculados de primer año hombres como faltantes")
print("2.- Se filtra por la columna total matriculados primer año distinta a valor faltante")
print("Lo que nos permitirá verificar si el valor faltante de hombres significa que todas las matriculas de primer año para estas condiciones son de mujeres.")
data_hombres_PA = matriculados[(matriculados['MATRICULADOS HOMBRES PRIMER AÑO'].isna()) & (~matriculados['TOTAL MATRICULADOS PRIMER AÑO'].isna())][['TOTAL MATRICULADOS PRIMER AÑO','MATRICULADOS MUJERES PRIMER AÑO']]
data_hombres_PA['dif'] = data_hombres_PA['TOTAL MATRICULADOS PRIMER AÑO'] - data_hombres_PA['MATRICULADOS MUJERES PRIMER AÑO']
print("Se establecen las condiciones y se seleccionan las columnas de interes")
print("Se crea la columna dif, que es la diferencia entre el total y la columnas de mujeres")
print("y el resultado es dif: ")
display(data_hombres_PA['dif'].sum())

Se establecen dos condiciones para verificar la integridad de los datos:
1.- Se filtra por todos los matriculados de primer año hombres como faltantes
2.- Se filtra por la columna total matriculados primer año distinta a valor faltante
Lo que nos permitirá verificar si el valor faltante de hombres significa que todas las matriculas de primer año para estas condiciones son de mujeres.
Se establecen las condiciones y se seleccionan las columnas de interes
Se crea la columna dif, que es la diferencia entre el total y la columnas de mujeres
y el resultado es dif: 


0.0

### Conclusión 2.

Se verifica lo siguiente, en los casos en que la columna de hombres es `NA` la columna de total matriculados primer año y matriculados de mujeres en primer año son idénticas.

Se verifican si las sumatorias de matriculados mujeres y hombres es el mismo que el total de matriculados (De primer año)

Por último se asume que aquellas columnas en donde las 3 columnas son valores faltantes es porque el programa se dejo de impartir o no tiene matricula vigente.


## Análisis 3:

Se verificarán los valores faltantes de la columna ``modalidad``

In [10]:
print("Columna Modalidad:")
display(matriculados['MODALIDAD'].value_counts())

Columna Modalidad:


Presencial        161458
No Presencial       6229
Semipresencial      6166
Name: MODALIDAD, dtype: int64

In [11]:
print("Se filtra base de datos en donde MODALIDAD sea un valor faltante:")
matriculados_modalidadNA = matriculados[matriculados['MODALIDAD'].isna()]
display(matriculados_modalidadNA)


Se filtra base de datos en donde MODALIDAD sea un valor faltante:


AÑO  TOTAL MATRICULADOS  MATRICULADOS MUJERES POR PROGRAMA  \
173853  MAT_2010                   1                                NaN   
173854  MAT_2010                   2                                NaN   
173855  MAT_2010                   1                                NaN   
173856  MAT_2010                   1                                NaN   
173857  MAT_2010                   1                                NaN   
...          ...                 ...                                ...   
216148  MAT_2007                  18                           13.00000   
216149  MAT_2007                  13                            7.00000   
216150  MAT_2007                  16                           10.00000   
216151  MAT_2007                 123                           61.00000   
216152  MAT_2007                 169                          126.00000   

        MATRICULADOS HOMBRES POR PROGRAMA  TOTAL MATRICULADOS PRIMER AÑO  \
173853                            1.00000                            NaN   
173854                            2.00000                            NaN   
173855                            1.00000                            NaN   
173856                            1.00000                            NaN   
173857                            1.00000                            NaN   
...                                   ...                            ...   
216148                            5.00000                       18.00000   
216149                            6.00000                       13.00000   
216150                            6.00000                            NaN   
216151                           62.00000                       44.00000   
216152                           43.00000                       35.00000   

        MATRICULADOS MUJERES PRIMER AÑO  MATRICULADOS HOMBRES PRIMER AÑO  \
173853                              NaN                              NaN   
173854                              NaN                              NaN   
173855                              NaN                              NaN   
173856                              NaN                              NaN   
173857                              NaN                              NaN   
...                                 ...                              ...   
216148                         13.00000                          5.00000   
216149                          7.00000                          6.00000   
216150                              NaN                              NaN   
216151                         23.00000                         21.00000   
216152                         30.00000                          5.00000   

       CLASIFICACIÓN INSTITUCIÓN NIVEL 1 CLASIFICACIÓN INSTITUCIÓN NIVEL 2  \
173853          Institutos Profesionales          Institutos Profesionales   
173854          Institutos Profesionales          Institutos Profesionales   
173855          Institutos Profesionales          Institutos Profesionales   
173856          Institutos Profesionales          Institutos Profesionales   
173857          Institutos Profesionales          Institutos Profesionales   
...                                  ...                               ...   
216148                     Universidades            Universidades Privadas   
216149                     Universidades            Universidades Privadas   
216150                     Universidades            Universidades Privadas   
216151                     Universidades            Universidades Privadas   
216152                     Universidades            Universidades Privadas   

       CLASIFICACIÓN INSTITUCIÓN NIVEL 3  ...  TES PARTICULAR SUBVENCIONADO  \
173853          Institutos Profesionales  ...                           NaN   
173854          Institutos Profesionales  ...                           NaN   
173855          Institutos Profesionales  ...                           NaN   
173856          Institutos Profes

In [12]:
print("Recuento de valores faltantes según el año:")
display(matriculados_modalidadNA['AÑO'].value_counts())
print("Proporción de Modalidad según Base de Datos de Matriculados:")
display(round(matriculados['MODALIDAD'].value_counts(normalize=True)*100,2))

Recuento de valores faltantes según el año:


MAT_2010    12147
MAT_2009    11414
MAT_2008     9943
MAT_2007     8796
Name: AÑO, dtype: int64

Proporción de Modalidad según Base de Datos de Matriculados:


Presencial       92.87000
No Presencial     3.58000
Semipresencial    3.55000
Name: MODALIDAD, dtype: float64

### Conclusión N°3:

Dado el N° de Faltantes y el año que corresponden, se determina lo siguiente:

el 100% de los datos faltantes corresponden a datos de años entre el 2007 y 2010, por lo que podemos asumir que no existia la modalidad `No presencial` y `Semi presencial`.

Se sugiere reemplazar todos estos valores faltantes por la categoria "Presencial", que es lo que se asume a priori, o en su defecto distribuir de acuerdo a la proporción de la base de datos completa.

### Analisis 4:

Se verificaran las columnas `DURACIÓN ESTUDIO CARRERA` Y `DURACIÓN TOTAL DE LA CARRERA`


In [13]:
data_na = matriculados[(matriculados['DURACIÓN ESTUDIO CARRERA'].isna() & matriculados['DURACIÓN TOTAL DE CARRERA'].isna())]

In [14]:
data_na[['DURACIÓN ESTUDIO CARRERA','DURACIÓN TOTAL DE CARRERA']].isna().sum()

DURACIÓN ESTUDIO CARRERA     106
DURACIÓN TOTAL DE CARRERA    106
dtype: int64

In [15]:
print("Recuento de Duración de Estudio Carrera:")
display(matriculados['DURACIÓN ESTUDIO CARRERA'].value_counts())
print("Proporción de Duración de Estudio Carrera:")
display(matriculados['DURACIÓN ESTUDIO CARRERA'].value_counts(normalize=True)*100)

Recuento de Duración de Estudio Carrera:


5.00000     47672
4.00000     46061
8.00000     45862
10.00000    27953
9.00000     10345
2.00000      9388
6.00000      8818
3.00000      5598
1.00000      4889
12.00000     4524
7.00000      2131
11.00000     2067
14.00000      502
20.00000      170
24.00000       32
13.00000       21
16.00000       11
18.00000        1
33.00000        1
15.00000        1
Name: DURACIÓN ESTUDIO CARRERA, dtype: int64

Proporción de Duración de Estudio Carrera:


5.00000    22.06557
4.00000    21.31990
8.00000    21.22779
10.00000   12.93839
9.00000     4.78831
2.00000     4.34535
6.00000     4.08152
3.00000     2.59110
1.00000     2.26293
12.00000    2.09399
7.00000     0.98636
11.00000    0.95674
14.00000    0.23236
20.00000    0.07869
24.00000    0.01481
13.00000    0.00972
16.00000    0.00509
18.00000    0.00046
33.00000    0.00046
15.00000    0.00046
Name: DURACIÓN ESTUDIO CARRERA, dtype: float64

### Conclusión N°4:

Dado que la `DURACIÓN ESTUDIO CARRERA` y `DURACIÓN TOTAL DE CARRERA` son valores faltantes, y al ser menos del 0.1 % de los datos (106 recuentos), se decide eliminar las columnas asociadas a esta característica.


## Análisis N°5:

En el siguiente apartado se verificarán las columnas:

* ``RANGO DE EDAD 15 A 19 AÑOS``
* ``RANGO DE EDAD 20 A 24 AÑOS``
* ``RANGO DE EDAD 25 A 29 AÑOS``
* ``RANGO DE EDAD 30 A 34 AÑOS``
* ``RANGO DE EDAD 35 A 39 AÑOS``
* ``RANGO DE EDAD 40 Y MÁS AÑOS``
* ``RANGO DE EDAD SIN INFORMACIÓN``


In [16]:
data_edad = matriculados[['RANGO DE EDAD 15 A 19 AÑOS','RANGO DE EDAD 20 A 24 AÑOS','RANGO DE EDAD 25 A 29 AÑOS','RANGO DE EDAD 30 A 34 AÑOS','RANGO DE EDAD 35 A 39 AÑOS','RANGO DE EDAD 40 Y MÁS AÑOS','RANGO DE EDAD SIN INFORMACIÓN']]


In [17]:
print("Base de datos de Edades:")
display(data_edad.head())

Base de datos de Edades:


RANGO DE EDAD 15 A 19 AÑOS  RANGO DE EDAD 20 A 24 AÑOS  \
0                         NaN                         NaN   
1                         NaN                     6.00000   
2                         NaN                    15.00000   
3                         NaN                     8.00000   
4                         NaN                    58.00000   

   RANGO DE EDAD 25 A 29 AÑOS  RANGO DE EDAD 30 A 34 AÑOS  \
0                         NaN                     2.00000   
1                     1.00000                     2.00000   
2                     8.00000                     4.00000   
3                     2.00000                         NaN   
4                     6.00000                     7.00000   

   RANGO DE EDAD 35 A 39 AÑOS  RANGO DE EDAD 40 Y MÁS AÑOS  \
0                         NaN                          NaN   
1                         NaN                      1.00000   
2                     1.00000                      6.00000   
3                         NaN                          NaN   
4                     1.00000                      4.00000   

   RANGO DE EDAD SIN INFORMACIÓN  
0                            NaN  
1                            NaN  
2                            NaN  
3                            NaN  
4                            NaN

In [18]:
print("Corroboración en los casos en que todas las filas las columnas seleccionadas tengan valores na:")
display(data_edad.isna().all(axis=1).value_counts())

Corroboración en los casos en que todas las filas las columnas seleccionadas tengan valores na:


False    216153
dtype: int64

### Conclusión N°5:

Debido a ser rangos de edad y la Universidad una institución donde puedes entrar a cualquier edad, se pueden dar estos fenomenos que generen diferentes alternativas, por ende es posible que en algunos casos algunos rangos etarios no tengan matricula asociada, por ende se reemplazaran los valores por 0.

Adicional a esto, se corroboró si alguna fila todos los rangos tenian datos `nan`, y el código diseñado para verificar esto arrojo un **FALSE**.

## Análisis 6

Las columnas de análisis son las siguientes:

* ``PROMEDIO EDAD CARRERA``
* ``PROMEDIO EDAD MUJER``
* ``PROMEDIO EDAD HOMBRE``

y se cruzarán datos con `MATRICULADOS MUJERES TOTAL` Y `MATRICULADOS HOMBRES TOTAL`

In [19]:
matriculados.columns

Index(['AÑO', 'TOTAL MATRICULADOS', 'MATRICULADOS MUJERES POR PROGRAMA',
       'MATRICULADOS HOMBRES POR PROGRAMA', 'TOTAL MATRICULADOS PRIMER AÑO',
       'MATRICULADOS MUJERES PRIMER AÑO', 'MATRICULADOS HOMBRES PRIMER AÑO',
       'CLASIFICACIÓN INSTITUCIÓN NIVEL 1',
       'CLASIFICACIÓN INSTITUCIÓN NIVEL 2',
       'CLASIFICACIÓN INSTITUCIÓN NIVEL 3', 'CÓDIGO DE INSTITUCIÓN',
       'NOMBRE INSTITUCIÓN', 'ACREDITACIÓN INSTITUCIONAL', 'REGIÓN',
       'PROVINCIA', 'COMUNA', 'NOMBRE SEDE', 'NOMBRE CARRERA',
       'ÁREA DEL CONOCIMIENTO', 'CINE-F 1997 ÁREA', 'CINE-F 1997 SUBAREA',
       'ÁREA CARRERA GENÉRICA', 'CINE-F 2013 ÁREA', 'CINE-F 2013 SUBAREA',
       'NIVEL GLOBAL', 'CARRERA CLASIFICACIÓN NIVEL 1',
       'CARRERA CLASIFICACIÓN NIVEL 2', 'MODALIDAD', 'JORNADA',
       'TIPO DE PLAN DE LA CARRERA', 'DURACIÓN ESTUDIO CARRERA',
       'DURACIÓN TOTAL DE CARRERA', 'CÓDIGO CARRERA', 'ACREDITACIÓN CARRERA',
       'TOTAL RANGO DE EDAD', 'RANGO DE EDAD 15 A 19 AÑOS',
       'RAN

In [20]:
data_6 = matriculados[[' PROMEDIO EDAD CARRERA ',' PROMEDIO EDAD MUJER ',' PROMEDIO EDAD HOMBRE ','MATRICULADOS MUJERES POR PROGRAMA','MATRICULADOS HOMBRES POR PROGRAMA','RANGO DE EDAD SIN INFORMACIÓN']]

In [21]:
print("Cuando promedio de edad de carrera es igual a nan, se debe a que existen rangos de Edad sin Información, por ende no se puede determinar correctamente el promedio")
display(data_6[(data_6[' PROMEDIO EDAD CARRERA '].isna())].head())

Cuando promedio de edad de carrera es igual a nan, se debe a que existen rangos de Edad sin Información, por ende no se puede determinar correctamente el promedio


PROMEDIO EDAD CARRERA   PROMEDIO EDAD MUJER   PROMEDIO EDAD HOMBRE   \
55157                     NaN                   NaN                    NaN   
55577                     NaN                   NaN                    NaN   
57182                     NaN                   NaN                    NaN   
59035                     NaN                   NaN                    NaN   
73482                     NaN                   NaN                    NaN   

       MATRICULADOS MUJERES POR PROGRAMA  MATRICULADOS HOMBRES POR PROGRAMA  \
55157                           18.00000                            5.00000   
55577                                NaN                            1.00000   
57182                            1.00000                                NaN   
59035                                NaN                           13.00000   
73482                                NaN                            1.00000   

       RANGO DE EDAD SIN INFORMACIÓN  
55157                       23.00000  
55577                        1.00000  
57182                        1.00000  
59035                       13.00000  
73482                        1.00000

In [22]:
print("Cuando el promedio de edad mujer es igual a nan, y tanto el n° de matriculados de mujeres como la edad promedio de la carrera son distintos a nan")
print("se entiende que el N° de mujeres matriculadas por programa se enlazan con los de rango de edad sin información.")
display(data_6[(matriculados[' PROMEDIO EDAD MUJER '].isna()) & (~matriculados['MATRICULADOS MUJERES POR PROGRAMA'].isna()) & (~matriculados[' PROMEDIO EDAD CARRERA '].isna())].head())

Cuando el promedio de edad mujer es igual a nan, y tanto el n° de matriculados de mujeres como la edad promedio de la carrera son distintos a nan
se entiende que el N° de mujeres matriculadas por programa se enlazan con los de rango de edad sin información.


PROMEDIO EDAD CARRERA   PROMEDIO EDAD MUJER   PROMEDIO EDAD HOMBRE   \
101120                   40,8                    NaN                  40,8    
103126                   56,0                    NaN                  56,0    
117937                   55,0                    NaN                  55,0    
124439                   42,0                    NaN                  42,0    
124440                   27,5                    NaN                  27,5    

        MATRICULADOS MUJERES POR PROGRAMA  MATRICULADOS HOMBRES POR PROGRAMA  \
101120                            1.00000                            8.00000   
103126                            1.00000                            1.00000   
117937                            1.00000                            1.00000   
124439                           13.00000                           14.00000   
124440                           41.00000                           69.00000   

        RANGO DE EDAD SIN INFORMACIÓN  
101120                        1.00000  
103126                        1.00000  
117937                        1.00000  
124439                       26.00000  
124440                      108.00000

In [23]:
print("Lo mismo para el caso de los hombres")
display(data_6[(matriculados[' PROMEDIO EDAD HOMBRE '].isna()) & (~matriculados['MATRICULADOS HOMBRES POR PROGRAMA'].isna()) & (~matriculados[' PROMEDIO EDAD CARRERA '].isna())].head())

Lo mismo para el caso de los hombres


PROMEDIO EDAD CARRERA   PROMEDIO EDAD MUJER   PROMEDIO EDAD HOMBRE   \
58592                   27,0                  27,0                     NaN   
64736                   25,5                  25,5                     NaN   
64737                   28,4                  28,4                     NaN   
64776                   30,5                  30,5                     NaN   
67046                   50,0                  50,0                     NaN   

       MATRICULADOS MUJERES POR PROGRAMA  MATRICULADOS HOMBRES POR PROGRAMA  \
58592                            4.00000                            3.00000   
64736                          377.00000                            3.00000   
64737                          328.00000                            1.00000   
64776                           31.00000                            1.00000   
67046                            1.00000                            1.00000   

       RANGO DE EDAD SIN INFORMACIÓN  
58592                        6.00000  
64736                        3.00000  
64737                        1.00000  
64776                        1.00000  
67046                        1.00000

### Conclusión 6:

Debido a lo anterior, se puede concluir que se pueden reemplazar los valores nan por 0. debido a que la mayoria de los casos depende por el rango de edad sin información, lo cuál es un número indeterminado.

## Análisis 7:

A continuación se analizarán las siguientes columnas:


* ``TES MUNICIPAL ``                          
* ``TES PARTICULAR SUBVENCIONADO  ``           
* ``TES PARTICULAR PAGADO  ``                 
* ``TES CORP. DE ADMINISTRACIÓN DELEGADA  ``  
* ``TES SERVICIO LOCAL EDUCACION``            

En relación con:

* `TOTAL TES`

In [24]:

data_7 = matriculados[['TES MUNICIPAL','TES PARTICULAR SUBVENCIONADO','TES PARTICULAR PAGADO','TES CORP. DE ADMINISTRACIÓN DELEGADA','TES SERVICIO LOCAL EDUCACION','TOTAL TES']]

In [25]:
columnas_tes = ['TES MUNICIPAL','TES PARTICULAR SUBVENCIONADO','TES PARTICULAR PAGADO','TES CORP. DE ADMINISTRACIÓN DELEGADA','TES SERVICIO LOCAL EDUCACION']

In [26]:
data_7['comprobacion_total'] = data_7[columnas_tes].sum(axis=1).astype(int)

C:\Users\aigna\AppData\Local\Temp\ipykernel_24328\3725993545.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_7['comprobacion_total'] = data_7[columnas_tes].sum(axis=1).astype(int)


In [27]:
data_7['dif'] = data_7['TOTAL TES'] - data_7['comprobacion_total']

C:\Users\aigna\AppData\Local\Temp\ipykernel_24328\3451614212.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_7['dif'] = data_7['TOTAL TES'] - data_7['comprobacion_total']


In [28]:
data_7['dif'].sum()

0

### Conclusión 7

Se puede reemplazar los valores faltantes por 0, debido a que corresponden a la sumatoria total de `TOTAL TES`, dado que los na no suman, no afecta si se ponen elementos de valor 0 al reemplazar.

## Análisis 8:

Se analizarán las siguientes variables:

* `TIPO ESTABLECIMIENTO HC`
* `TIPO ESTABLECIMIENTO TP`
* `CLAS_EST ADULTO`
* `CLAS_EST JOVEN`


Se identifica que las dos primeras columnas se relacionan con las dos ultimas, es decir ambos pares suman la totalidad, por ende se sumaran las columnas y se compararán los totales.

In [30]:
dat_8 = matriculados[['TIPO ESTABLECIMIENTO HC','TIPO ESTABLECIMIENTO TP','CLAS_EST ADULTO','CLAS_EST JOVEN']]

In [34]:
list1 = ['TIPO ESTABLECIMIENTO HC','TIPO ESTABLECIMIENTO TP']
list2 = ['CLAS_EST ADULTO','CLAS_EST JOVEN']

In [ ]:
dat_8['suma1']= dat_8[list1].sum(axis=1)
dat_8['suma2'] = dat_8[list2].sum(axis=1)

In [ ]:
dat_8['dif'] = dat_8['suma1']-dat_8['suma2']

In [49]:
dat_8['dif'].sum()

0.0

### Conclusión 8:

Se determina que reemplazar los valores faltantes por `nan`, no afectan la integridad de los datos.